### DISCUSSION TITLE
Change value of gravity in environment

### DISCUSSION THREAD

**Discussion by harshitsankhla** [microsoft/AirSim#4217] [Support Q&A] [answered]
### What's your question?
Any way to change the value of gravity in the environment state

The airsim client has a method to retrieve the current environment state - https://microsoft.github.io/AirSim/api_docs/html/index.html#airsim.types.EnvironmentState

Any way we can change this in real-time or before we enable the client ? Say I want to change the value of gravity to some other value. 

### Include context on what you are trying to achieve

Want to change the value of gravity to observe some specific effects in a velocity controller.

### Include details of what you already did to find answers
getter method available in client object as - simGetGroundTruthEnvironment, cannot find a setter method

**Comment by jonyMarino** [comment, marked as answer]
Hi and welcome @harshitsankhla! We don't have an API to accomplish that. You will need to change the AirSim code. [Here](https://github.com/microsoft/AirSim/blob/4266b898ac5b20f35ebca3f2e3fa821b336dec69/AirLib/include/physics/Environment.hpp#L119) you can find the place where the gravity is calculated. It can be interesting to have setters for the environments; you can submit a pull request.

**Reply by harshitsankhla** [reply]
Hi! Thanks for getting back on this. Would love to work on a PR if this can be a useful feature for others as well. I will post updates here.
